### DISCUSSION TITLE
Separate post content from things like tagged accounts, added links, hashtags

### DISCUSSION THREAD

**Discussion by venteto** [bluesky-social/atproto#2029] [Bluesky Lexicons]
Ideally these sorts of quasi-metadata would never impact the character count quota of the actual content of the posts

As an example, if you want post a link to an older article published elsewhere by an organization that, while they may have an account on the same social network, never actually created a social post of their own as a referral to the off-site article, it would be nice to be able to nevertheless tag their in-network social account handle, as a credit / citation of them as a source. At the same time you need to be able to include enough text from the article, your own separate commentary on it, or both, to make it apparent as to why the article might be worth reading, and thus the in-post link to it worthy of clicking. All of these aspects of the post compete for the finite number of characters avaiable in a quota-based system. Moreover, making a "thread" with an additional post to merely cite the account and include a link is just awkward, unnatural, and interrupting of the flow of content, content that would still be concise in one post, if the post were not fettered by links, account tagging, and hashtagging consuming characters that could be used for actual post _content_.

Ideally the the protocol and client app would not repeat the failings of Twitter, and this was one of the negatives of Twitter (and even Mastodon). This is a greenfield opportunity to do something that supersedes that old model and impoves upon it, rather than mostly clones it.

**Comment by DavidBuchanan314** [comment]
Isn't this exactly what the facet system achieves?

**Reply by venteto** [reply]
Look at this example. I've already used almost 50 characters of my quota, and I haven't even written a post yet. The facet system did not preserve 300 characters for me to use for the actual post content.

![Screenshot from 2024-01-08 09-42-49](https://github.com/bluesky-social/atproto/assets/7062385/bf3fbc22-0926-47a9-bbdc-3c7a70dc4e1b)

**Reply by pfrazee** [reply]
FWIW hashtags can work externally (once that all goes live) and links can be shortened via richtext or as link cards. That just leaves handles
